### PR TITLE
add root api status page

### PR DESCRIPTION
**Pull request by mabiede**

### CODE REVIEW

File: pool/queue/repo.ml [hunk 1]
```diff
@@ -160,18 +160,24 @@ let find label id =
   ||> CCOption.to_result Pool_common.Message.(NotFound Field.Queue)
 ;;
 
+let workable_query =
```
Comment (timohuber) [inline pool/queue/repo.ml:163]: That could directly be integrated in `find_workable_request`

File: pool/routes/routes.ml [hunk 2]
```diff
@@ -789,6 +789,7 @@ module Root = struct
     ; post "/request-reset-password" Login.request_reset_password_post
     ; get "/reset-password" Login.reset_password_get
     ; post "/reset-password" Login.reset_password_post
+    ; get "/api/v1/status" Status.show
```
Comment (timohuber) [inline pool/routes/routes.ml:792]: IMO the status does not need a version namespace.

**Review by timohuber — CHANGES_REQUESTED** [review]

File: pool/queue/repo.ml (hunk 1 above)
Comment (mabiede) [reply pool/queue/repo.ml]: this one is used by the count at the file end as well. I combined the query generation.

File: pool/routes/routes.ml (hunk 2 above)
Comment (mabiede) [reply pool/routes/routes.ml:792]: as discussed in channel

File: pool/queue/repo.ml [hunk 3]
```diff
@@ -160,19 +160,24 @@ let find label id =
   ||> CCOption.to_result Pool_common.Message.(NotFound Field.Queue)
 ;;
 
-let find_workable_request =
-  let open Caqti_request.Infix in
+let find_workable_query ?(count = false) () =
+  let select =
+    if count then "SELECT COUNT (*) FROM queue_jobs" else select_from_fragment
```
Comment (timohuber) [inline pool/queue/repo.ml:165]: apparently this is a syntax error 😩 

```suggestion
    if count then "SELECT COUNT(*) FROM queue_jobs" else select_from_fragment
```

File: pool/web/handler/root_status.ml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+       ( Label.value label
+       , CCOption.map_or ~default:"unknown" LastRunAt.show last_run )))
+    @ [ "job_count", CCInt.to_string job_count
+      ; ("ok", if is_ok schedules then "true" else "false")
+      ]
+    |> CCList.map (fun (k, v) -> `Tuple [ `String k; `String v ])
+    |> Lwt.return_ok
+  in
+  match result with
+  | Ok result -> yojson_response (`List result)
```
Comment (timohuber) [inline pool/web/handler/root_status.ml:31]: Formatting JSON, from 
```[("system_events","2023-12-19 07:59:13 +00:00"),("job_count","0"),("ok","true")]```
to 
```
{
   "system_events":"2023-12-19 08:04:06 +00:00",
   "job_count":"0",
   "ok":"true"
}
```

```suggestion
    |> CCList.map (fun (k, v) -> k, `String v)
    |> Lwt.return_ok
  in
  match result with
  | Ok result -> yojson_response (`Assoc result)
```

**Review by timohuber — CHANGES_REQUESTED** [review]

**Review by timohuber — APPROVED** [review]

**Pull request merged**